libtiny_tui/src/line_split.rs

```diff
@@ -0,0 +1,152 @@
+        let mut temp_count = 1;
+        if let Some(line_count) = self.line_count {
+            temp_count = line_count;
+            // If we made space for the cursor, subtract it.
+            if self.has_cursor && self.current_line_length == self.line_width {
+                temp_count -= 1;
+            }
+        }
+        for (c, current_idx) in buffer.skip(offset).zip(offset..) {
+            let current_idx = current_idx as i32;
+            self.current_line_length += 1;
+
+            if c.is_whitespace() {
+                // Splitting
+                if self.current_line_length > self.line_width {
+                    // we're on a whitespace so just go to next line
+                    temp_count += 1;
+                    // this character will be the first one on the next line
+                    self.current_line_length = 1;
+                    // nick is shown on the first line, set width to full width in the consecutive
+                    // lines
+                    self.line_width = self.width;
+                    // store index for drawing
+                    self.split_indices.push(current_idx);
+                }
+                // store whitespace for splitting
+                self.last_whitespace_idx = Some(current_idx);
+                self.prev_char_is_whitespace = true;
+            } else {
+                // Splitting
+                if self.current_line_length > self.line_width {
+                    // if the previous character was a whitespace, then we have a clean split
+                    if !self.prev_char_is_whitespace && self.last_whitespace_idx.is_some() {
+                        // move back to the last whitespace and get the length of the input that
+                        // will be on the next line
+                        self.current_line_length = current_idx - self.last_whitespace_idx.unwrap();
+                        // store index for drawing
+                        self.split_indices
+                            .push(self.last_whitespace_idx.unwrap() + 1);
+                    } else {
+                        // unclean split on non-whitespace
+                        self.current_line_length = 1;
+                        // store index for drawing
+                        self.split_indices.push(current_idx);
+                    }
+                    // invalidate whitespace since we split here
+                    self.last_whitespace_idx = None;
+                    // moved to next line
+                    temp_count += 1;
+                    // set width to full width
+                    self.line_width = self.width;
+                }
+                self.prev_char_is_whitespace = false;
+            }
+        }
+
+        // Last line length is `line_width`, make room for cursor
+        if self.has_cursor && self.current_line_length == self.line_width {
+            temp_count += 1;
+        }
+        self.line_count = Some(temp_count);
+    }
+}
```

Comment:
This is moving old code around, are there any changes or is this identical to the moved version? `has_cursor` is new, are there any other changes?

Reply:
This is all identical except the addition of not having a cursor to account for.